lib/explorer/polars_backend/data_frame.ex

```diff
@@ -33,7 +33,7 @@ defmodule Explorer.PolarsBackend.DataFrame do
     dtypes =
       if dtypes do
         Enum.map(dtypes, fn {colname, dtype} ->
-          {Atom.to_string(colname), Shared.internal_from_dtype(dtype)}
+          {colname, Shared.internal_from_dtype(dtype)}
```

Comment:
It is probably worth adding some validation here (check the types and raise invalid dtype otherwise).